src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,433 @@
+		List<DemandeAvenantPivot> demandes = getDemandesAvenantsDunContrat(refContrat);
+		return demandes.stream().filter(neCorrespondPasAUnEtatTerminal()).collect(Collectors.toList());
+	}
+
+	public Optional<DemandeAvenantPivot> getDemandeAvenantEnCours(String refContrat) {
+		List<DemandeAvenantPivot> demandesOuvertes = getDemandesAvenantsEnCours(refContrat).stream().filter(neCorrespondPasAUnEtatTerminal()).collect(Collectors.toList());
+		if (demandesOuvertes.size() > 1) {
+			throw new ComponentCoherenceDatasException(
+					"Il existe plusieurs demandes avenants ouvertes : " + demandesOuvertes.stream().map(DemandeAvenantPivot::getId).collect(Collectors.toList()).toString());
+		}
+		if (demandesOuvertes.isEmpty()) {
+			return Optional.empty();
+		}
+		return Optional.ofNullable(demandesOuvertes.get(0));
+
+	}
+
+	public List<DemandeAvenantPivot> getDemandesAvenantsAcceptees(String refContrat) {
+		List<DemandeAvenantEntity> entities = getEntitiesDemandesAvenantsDunContrat(refContrat);
+
+		// @formatter:off
+		return entities.stream().filter(Objects::nonNull)
+				.filter(garderLaPhotoAvenantAccepte())
+				.map(e -> convDemandeAvenant.convertEntityToPivot(e))
+				.filter(Optional::isPresent)
+				.map(Optional::get)
+				.sorted(Comparator.comparing(DemandeAvenantPivot::getId)).collect(Collectors.toList());
+		// @formatter:on
+	}
+
+	public List<DemandeAvenantPivot> getDemandesAvenantsEnCoursNonModifiablesMaisAnnulables(String refContrat) {
+		List<DemandeAvenantPivot> demandes = getDemandesAvenantsEnCours(refContrat);
+		return demandes.stream().filter(e -> {
+			try {
+				return e.getEtatEnum().isPresent() && !e.getEtatEnum().get().estModifiable();
+			} catch (EnumNotFoundException e1) {
+				return false;
+			}
+		}).filter(e -> {
+			try {
+				return e.getEtatEnum().isPresent() && e.getEtatEnum().get().estAnnulableParIntermediaire();
+			} catch (EnumNotFoundException e1) {
+				return false;
+			}
+		}).collect(Collectors.toList());
+	}
+
+	public List<DemandeAvenantPivot> getDemandesAvenantsDansEtatTerminal(String refContrat) {
+		List<DemandeAvenantPivot> demandes = getDemandesAvenantsDunContrat(refContrat);
+		return demandes.stream().filter(correspondAUnEtatTerminal()).collect(Collectors.toList());
+	}
+
+	private Predicate<? super DemandeAvenantPivot> neCorrespondPasAUnEtatTerminal() {
+		return correspondAUnEtatTerminal().negate();
+	}
+
+	private Predicate<? super DemandeAvenantPivot> correspondAUnEtatTerminal() {
+		return e -> EtatDemandeAvenant.getByCode(e.getEtat().getCode()).estTerminal();
+	}
+
+	public Optional<DemandeAvenantPivot> getDemandeAvenant(BigInteger idDemandeAvenant) {
+		if (idDemandeAvenant != null) {
+			Optional<DemandeAvenantEntity> oEntity = repositoryDemandeAvenant.findById(idDemandeAvenant);
+			if (oEntity.isPresent()) {
+				return convDemandeAvenant.convertEntityToPivot(oEntity.get());
+			}
+		}
+		return Optional.empty();
+	}
+
+	@Deprecated
```

Comment:
Deprecated Method - Deprecated methods should be documented and ideally removed to avoid confusion and encourage the use of updated alternatives. (Severity: low)